newsroom/settings.py

```diff
@@ -23,10 +24,11 @@ def get_settings_collection():
 
 @blueprint.route("/settings/<app_id>")
 @account_manager_only
-def app(app_id):
+async def app(app_id):
     for app in flask.current_app.settings_apps:
         if app._id == app_id:
-            return flask.render_template("settings.html", setting_type=app_id, data=app.data())
+            data = await app.data() if iscoroutine(app.data()) else app.data()
```

Comment:
Here you're calling `app.data()` twice, once for assignment and the other for if condition.
It would be better to call it once, and check the response value in the condition